WordPress/wordpress-playground
# Fixing Build Regression [BISON]

## What is this PR doing?

This PR corrects the same Bison2.7 issue as https://github.com/WordPress/wordpress-playground/pull/871, in a much simpler fashion.

## What problem is it solving?

Bison2.7 will not be built in some cases, blocking the build of PHP <= 7.3.

## How is the problem addressed?

This invokes dependency builds before building PHP.

## Testing Instructions

Checkout the branch & install the dependencies:

```
git checkout sm-make-deps-before-compile
npm install
```

In a separate terminal, clear the artifacts:

```bash
docker system prune -af
cd packages/php-wasm/compile
make clean
```

Run the build and ensure it succeeds:

```bash
npm run recompile:php:node
npm run recompile:php:web
```

Start the dev server:

```bash
npm run dev
```

Then, navigate to:

* http://localhost:5400/website-server/?php=8.3
* http://localhost:5400/website-server/?php=8.2
* http://localhost:5400/website-server/?php=8.1
* http://localhost:5400/website-server/?php=8.0
* http://localhost:5400/website-server/?php=7.4
* http://localhost:5400/website-server/?php=7.3
* http://localhost:5400/website-server/?php=7.2
* http://localhost:5400/website-server/?php=7.1
* http://localhost:5400/website-server/?php=7.0

... and ensure that wordpress works as-expected in each environment.
